packages/calcite-components/src/components/tab-nav/tab-nav.tsx

```diff
@@ -301,35 +412,197 @@ export class TabNav {
+
+  private activeIndicatorEl: HTMLElement;
 
-  parentTabsEl: HTMLCalciteTabsElement;
+  private activeIndicatorContainerEl: HTMLDivElement;
 
-  tabNavEl: HTMLDivElement;
+  private dir: Direction = "ltr";
 
-  activeIndicatorEl: HTMLElement;
+  private containerEl: HTMLDivElement;
 
-  activeIndicatorContainerEl: HTMLDivElement;
+  private parentTabsEl: HTMLCalciteTabsElement;
 
-  animationActiveDuration = 0.3;
+  private tabTitleContainerEl: HTMLDivElement;
+
+  private intersectionObserver: IntersectionObserver;
+
+  private resizeObserver = createObserver("resize", () => {
+    this.updateScrollingState();
 
-  resizeObserver = createObserver("resize", () => {
     if (!this.activeIndicatorEl) {
       return;
     }
 
-    // remove active indicator transition duration during resize to prevent wobble
-    this.activeIndicatorEl.style.transitionDuration = "0s";
-    this.updateActiveWidth();
-    this.updateOffsetPosition();
+    this.updateActiveIndicator();
   });
 
+  private get scrollerButtonWidth(): number {
+    const { scale } = this;
+    return parseInt(scale === "s" ? calciteSize32 : scale === "m" ? calciteSize48 : calciteSize64);
+  }
+
   //--------------------------------------------------------------------------
   //
   //  Private Methods
   //
   //--------------------------------------------------------------------------
 
+  private updateActiveIndicator(): void {
+    const tabTitleScrollLeft = this.tabTitleContainerEl?.scrollLeft;
+    const containerScrollLeft = this.containerEl?.scrollLeft;
+    const navWidth = this.activeIndicatorContainerEl?.offsetWidth;
+    const tabLeft = this.selectedTitle?.offsetLeft;
+    const tabWidth = this.selectedTitle?.offsetWidth;
+    const offsetRight = navWidth - tabLeft - tabWidth;
+    const offsetBase = this.dir === "ltr" ? tabLeft : offsetRight;
+    const multiplier = this.dir === "ltr" ? -1 : 1;
+
+    this.indicatorOffset = offsetBase + multiplier * (containerScrollLeft + tabTitleScrollLeft);
+    this.indicatorWidth = this.selectedTitle?.offsetWidth;
+  }
+
+  private onTabTitleWheel = (event: WheelEvent): void => {
+    event.preventDefault();
+    const scrollByX = (this.dir === "rtl" ? -1 : 1) * event.deltaY;
+    (event.currentTarget as HTMLDivElement).scrollBy(scrollByX, 0);
+    requestAnimationFrame(() => this.updateActiveIndicator());
+  };
+
+  private onSlotChange = (event: Event): void => {
+    this.intersectionObserver?.disconnect();
+
+    const slottedChildren = (event.target as HTMLSlotElement).assignedElements();
+    slottedChildren.forEach((child) => {
+      this.intersectionObserver?.observe(child);
+    });
+  };
+
+  private storeContainerRef = (el: HTMLDivElement) => (this.containerEl = el);
+
+  private storeTabTitleWrapperRef = (el: HTMLDivElement) => {
+    this.tabTitleContainerEl = el;
+
+    if (!el) {
+      return;
+    }
+
+    this.intersectionObserver = createObserver("intersection", () => this.updateScrollingState(), {
```

Comment:
Any chance `storeTabTitleWrapperRef` could be called again and this intersectionObserver may need to be disconnected before the new one is created?